app/code/Magento/ProductAlert/Block/Adminhtml/Catalog/Product/Edit/Tab/Alerts/Renderer/Stock.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Magento\ProductAlert\Block\Adminhtml\Catalog\Product\Edit\Tab\Alerts\Renderer;
+
+use Magento\Backend\Block\Widget\Grid\Column\Renderer\AbstractRenderer;
+use Magento\Framework\DataObject;
+use Magento\Framework\Message\ManagerInterface;
+use Magento\Store\Model\StoreManagerInterface;
+use Magento\Backend\Block\Context;
+use Magento\InventorySales\Model\StockResolver;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+
+class Stock extends AbstractRenderer
+{
+    /**
+     * @var StoreManagerInterface
+     */
+    protected $_storeManager;
+
+    /**
+     * @var StockResolver
+     */
+    protected $_stockResolver;
+
+    /**
+     * @var ManagerInterface
+     */
+    protected $_messageManager;
+
+    /**
+     * Stock constructor.
+     * @param Context $context
+     * @param StoreManagerInterface $storeManager
+     * @param StockResolver $stockResolver
+     * @param ManagerInterface $messageManager
+     * @param array $data
+     */
+    public function __construct(
+        Context $context,
+        StoreManagerInterface $storeManager,
+        StockResolver $stockResolver,
+        ManagerInterface $messageManager,
+        array $data = []
+    )
+    {
+        parent::__construct($context, $data);
+        $this->_storeManager = $storeManager;
+        $this->_stockResolver = $stockResolver;
+        $this->_messageManager = $messageManager;
+    }
+
+    /**
+     * Handles Stock column
+     *
+     * @param DataObject $row
+     * @return string
+     */
+    public function render(DataObject $row)
+    {
+        try {
+            $value = $this->getColumnData($row);
+            return $value;
+        } catch (\Exception $exception) {
+            $this->_messageManager->addErrorMessage($exception);
+        }
+        return parent::render($row);
+    }
+
+    /**
+     * @param $row
+     * @return string
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function getColumnData($row)
+    {
+        /** @var \Magento\Store\Api\Data\WebsiteInterface $website */
+        $website = $this->_storeManager->getWebsite($row->getWebsiteId());
+
+        if (!$website) {
+            return '';
+        }
+        $websiteStr = $this->getWebsiteStr($website);
+        $stockName = $this->getStockStr($website);
+        $resultStr = $websiteStr . '<br/>' . $stockName;
+        return $resultStr;
+    }
+
+    /**
+     * @param $website \Magento\Store\Api\Data\WebsiteInterface
+     * @return string
+     */
+    public function getWebsiteStr($website)
```

Comment:
Provide type hinting for all methods (parameter types, return types)

Reply:
I understood about parameter types, But about return types. Did you mean something like that? 
For example:
```php
function sum($a, $b): float {
    return $a + $b;
}
```

Reply:
Yes
```
function sum(float $a, float $b): float {
    return $a + $b;
}
```
We have only one restriction
We cannot provide additional type hinting if we have parent method without type hinting and we can not change parent legacy class because it will be backward incompatible. But for new code we try to use defensive programming